src/main/java/org/example/StudentRepository.java

```diff
@@ -0,0 +1,145 @@
+package org.example;
+
+
+import java.time.DateTimeException;
+import java.time.LocalDate;
+import java.util.*;
+
+
+public class StudentRepository {
+    protected HashMap<Integer, Student> studentHashMap = new HashMap<>();
+
+    protected void createStudent() {
+        boolean IdValidation = false;
+        boolean firstNameValidation = false;
+        boolean lastNameValidation = false;
+        boolean dateOfBirthValidation = false;
+        boolean genderValidation = false;
+        String genderInput;
+        String dateOfBirthInput;
+        Student createdStudent = new Student("a", "b", LocalDate.parse("2000-10-10"), "d", 0);
+        Scanner scanner = new Scanner(System.in);
+
+        System.out.println("Please enter Student first name: ");
```

Comment:
please don't read data from the keyboard, add students in the main class, don't use scanner. rewrite these without using booleans, just use the exception messages. Also group all the validation into one ore more methods that check new newly created student before adding it to your repo